src/command_modules/azure-cli-iot/azure/cli/command_modules/iot/custom.py

```diff
@@ -112,6 +138,38 @@ def iot_hub_policy_get(client, hub_name, policy_name, resource_group_name=None):
     return client.get_keys_for_key_name(resource_group_name, hub_name, policy_name)
 
 
+def iot_hub_policy_create(client, hub_name, policy_name, permissions, resource_group_name=None):
+    hub = iot_hub_get(client, hub_name, resource_group_name)
+    policies = iot_hub_policy_list(client, hub_name, hub.resourcegroup)
+    if _is_policy_existed(policies, policy_name):
+        raise CLIError('Policy {0} already existed.'.format(policy_name))
+    updated_policies = []
+    updated_policies.extend(policies)
```

Comment:
I am afraid it still doesn't work. Exception is thrown when serializing the policy object.
```
  File "d:\workspace\azure-cli-x\src\command_modules\azure-cli-iot\azure\cli\command_modules\iot\custom.py", line 164, in iot_hub_policy_create
    return client.create_or_update(hub.resourcegroup, hub_name, hub, {'IF-MATCH': hub.etag})
  File "D:\workspace\azure-cli-x\env\lib\site-packages\azure\mgmt\iothub\operations\iot_hub_resource_operations.py", line 154, in create_or_update
    body_content = self._serialize.body(iot_hub_description, 'IotHubDescription')
  File "D:\workspace\azure-cli-x\env\lib\site-packages\msrest\serialization.py", line 303, in body
    data = _convert_to_datatype(data, data_type, self.dependencies)
  File "D:\workspace\azure-cli-x\env\lib\site-packages\msrest\serialization.py", line 171, in _convert_to_datatype
    getattr(data, attr), map['type'], localtypes))
  File "D:\workspace\azure-cli-x\env\lib\site-packages\msrest\serialization.py", line 171, in _convert_to_datatype
    getattr(data, attr), map['type'], localtypes))
  File "D:\workspace\azure-cli-x\env\lib\site-packages\msrest\serialization.py", line 150, in _convert_to_datatype
    param, data_type[1:-1], localtypes) for param in data
  File "D:\workspace\azure-cli-x\env\lib\site-packages\msrest\serialization.py", line 150, in <listcomp>
    param, data_type[1:-1], localtypes) for param in data
  File "D:\workspace\azure-cli-x\env\lib\site-packages\msrest\serialization.py", line 164, in _convert_to_datatype
    localtypes) for key in data
  File "D:\workspace\azure-cli-x\env\lib\site-packages\msrest\serialization.py", line 164, in <dictcomp>
    localtypes) for key in data
TypeError: 'SharedAccessSignatureAuthorizationRulePaged' object is not subscriptable
```